packages/eas-cli/src/credentials/ios/actions/SetupBuildCredentials.ts

```diff
@@ -40,13 +69,17 @@ export class SetupBuildCredentials implements Action<IosAppBuildCredentials> {
     await ctx.bestEffortAppStoreAuthenticateAsync();
 
     if (ctx.appStore.authCtx) {
+      const enableAssociatedDomains = isAssociatedDomainsDefined(ctx.projectDir, ctx.exp);
```

Comment:
You don't know if exp is here or even if it is this interface does not guarantee that exp refers to the same app as `this.options.app`. I'm not sure if it's even possible to call this class in a way that would use different projects, but either way this code is not safe.

I'm not 100% how to handle this approach, you could do this by
```suggestion
      const enableAssociatedDomains = ctx.hasProjectContext && ctx.exp.slug === this.options.app.projectName ? isAssociatedDomainsDefined(ctx.projectDir, ctx.exp) : undefined;
```
and changing ensureBundleIdExistsAsync to not do anything in case of undefined (neither set nor unset)

Reply:
cc @quinlanj 
what do you think about it?

Reply:
1. On project context: This class is called by `build:ios` and the manager, afaik its always called in a project context, unless it's possible to call `build:ios` outside your project directory.
2. On `this.options.app` not referring to the same app as `ctx.exp`: this is possible but it was not a usecase I thought of 😅  We currently derive `this.options.app` from whatever is declared `ctx.exp`.

That being said, @wkozyra95  raises a great point and it's good to assert these assumptions in the class. I'd make these assertions at the beginning of `runAsync`, see here: